contracts/DEXes/UniV3StablesJoint.sol

```diff
@@ -0,0 +1,639 @@
+            _checkAllowance(address(_pool), IERC20(_tokenFrom), _amountIn);
+            uint256 prevBalance = IERC20(_tokenTo).balanceOf(address(this));
+            // Perform swap
+            _pool.exchange(
+                _getCRVPoolIndex(_tokenFrom, _pool), 
+                _getCRVPoolIndex(_tokenTo, _pool),
+                _amountIn, 
+                0
+            );
+            // Revoke allowance
+            IERC20(_tokenFrom).safeApprove(address(_pool), 0);
+            return (IERC20(_tokenTo).balanceOf(address(this)) - prevBalance);
+        }
+        
+    }
+
+    /*
+     * @notice
+     *  Function used internally to quote a potential rebalancing swap without actually 
+     * executing it. Same as the swap function, will simulate the trade either on the uni v3
+     * pool or CRV pool based on useCRVPool
+     * @param _tokenFrom, adress of token to swap from
+     * @param _tokenTo, address of token to swap to
+     * @param _amountIn, amount of _tokenIn to swap for _tokenTo
+     * @return simulated swapped amount
+     */
+    function quote(
+        address _tokenFrom,
+        address _tokenTo,
+        uint256 _amountIn
+    ) internal view override returns (uint256) {
+        require(_tokenTo == tokenA || _tokenTo == tokenB); // dev: must be a or b
+        require(_tokenFrom == tokenA || _tokenFrom == tokenB); // dev: must be a or b
+        if(!useCRVPool){
+            // Use uni v3 pool to swap
+            // Order of swap
+            bool zeroForOne = _tokenFrom < _tokenTo;
+
+            // Use the uniswap helper view to simluate the swapin the uni v3 pool
+            (int256 _amount0, int256 _amount1, , ) = UniswapHelperViews.simulateSwap(
+                // pool to use
+                IUniswapV3Pool(pool),
+                // order of swap
+                zeroForOne,
+                // amountSpecified
+                _amountIn.toInt256(),
+                // price limit
+                zeroForOne ? MIN_SQRT_RATIO + 1 : MAX_SQRT_RATIO - 1
+            );
+
+            // Ensure amounts are returned in right order and sign (uni returns negative numbers)
+            return zeroForOne ? uint256(-_amount1) : uint256(-_amount0);
+        } else {
+            // Do NOT use uni pool use CRV pool
+            
+            ICRVPool _pool = ICRVPool(crvPool);
+
+            // Call the quote function in CRV pool
+            return _pool.get_dy(
+                _getCRVPoolIndex(_tokenFrom, _pool), 
+                _getCRVPoolIndex(_tokenTo, _pool), 
+                _amountIn
+            );
+        }
+        
+    }
+
+    /*
+     * @notice
+     *  Function used internally to retrieve the CRV index for a token in a CRV pool, for example
+     * 3Pool uses:
+     * - 0 is DAI
+     * - 1 is USDC
+     * - 2 is USDT
+     * @return in128 containing the token's pool index
+     */
+    function _getCRVPoolIndex(address _token, ICRVPool _pool) internal view returns(int128) {
```

Comment:
if token doesn0t match any of the pool's tokens, it will return 0. 

maybe revert?

Reply:
Including it